ui/pages/confirmation/util.ts

```diff
@@ -0,0 +1,127 @@
+import { ResultComponent } from '@metamask/approval-controller';
+
+type TemplateRendererComponent = {
+  key: string;
+  element: string;
+  props?: Record<string, unknown>;
+  children?:
+    | string
+    | TemplateRendererComponent
+    | (string | TemplateRendererComponent)[];
+};
+
+export function processError(
+  input: undefined | string | ResultComponent | ResultComponent[],
+  fallback: string,
+): TemplateRendererComponent | (string | TemplateRendererComponent)[] {
+  let currentInput = convertResultComponents(input);
+
+  if (!currentInput) {
+    currentInput = fallback;
+  }
+
+  if (!(typeof currentInput === 'string')) {
+    return currentInput;
+  }
+
+  return {
+    key: 'error',
+    element: 'ActionableMessage',
+    props: { type: 'danger', message: currentInput },
+  };
+}
+
+export function processString(
```

Comment:
Shall we add some unit tests for this file?

The methods will be much easier to test in the abstract outside of a page, plus we'd only have to test once rather than for each result page.